### PR TITLE
Sample test and training subsets stochastically

### PR DESCRIPTION
**Pull request by Stektpotet**
The current implementation of `load_full_mnist()` loads a skewed subset of MNIST for validation, giving the impression that training is somehow doing way better on most unseen data as compared to the actual training data.

As is, the validation set consists of the 2000 _last samples_ of the original MNIST test set. Those 2000 samples do not span the _real_ domain represented by the full test set of 10000 samples. leading to these seemingly impressive validation accuracies and losses.

Optimally, students should use the full sets - but, as that may not be computationally ideal for students in general, I've made this change to how the subsets are sampled. The addition simply adds an option to sample stochastically without replacement from the whole original dataset (both training and test). I select `train_size` (defaults to 18000) indices in [0-59999] and `test_size` (defaults to 2000) indices in [0-9999] when stochastic sampling is turned on.
Otherwise the same sequential sampling is used as before, using training set indices [0-`train_size`] and [(10000-`test_size`)-9999].

Then these index-arrays are instead used to sample from the original datasets.

As with other random-dependent parts of the code, I have seeded the random to help in reproducibility for students.

|          |   Accuracy  progression  |
|-------|---------|
|  Sequential sub-set sampling (current state) |  task3b_softmax_train_accuracy    |
|   Stochastic sub-set sampling (suggested addition)   |  task3b_softmax_train_accuracy_stochastic |

|          |   Loss  progression  |
|-------|---------|
|  Sequential sub-set sampling (current state) |  task3b_softmax_train_loss    |
|   Stochastic sub-set sampling (suggested addition)   |  task3b_softmax_train_loss_stochastic |

All in all, this turned out to be a nice deep dive into the importance of what data we use and how. - kind of similar to the importance of data shuffling within mini-batch sampling. 

So maybe there's a lesson there for future students of the course too?

### CODE REVIEW

**Review by mbirkhez — APPROVED** [review]
Thanks for making the changes! It is definitely better to sample randomly :)

**Review by mbirkhez — APPROVED** [review]
